src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const updatedArray = sourceString
+    .split(';')
+    .map((item) => {
+      return item.split(':');
+    })
+    .filter((item) => item.length > 1)
+    .map((item) => {
+      return [item[0].trim(), item[1].trim()];
+    });
```

Comment:
[CODE PRACTICE] - Instead of using loops, you could use array iteration methods such as `reduce` to directly transform the array into an object, which would be more in line with functional programming practices.